src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -20952,32 +20952,1366 @@ public class C
     public C()
     {
         init();
+        field1.ToString();
+        field2.ToString();
+        field3.ToString(); // 1
+        field4.ToString(); // 2
+
+        [MemberNotNull(nameof(field1), nameof(field2))]
+        void init() => throw null!;
+    }
+}
+""", MemberNotNullAttributeDefinition], parseOptions: TestOptions.Regular9);
+
+            c.VerifyDiagnostics(
+                // (14,9): warning CS8602: Dereference of a possibly null reference.
+                //         field3.ToString(); // 1
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "field3").WithLocation(14, 9),
+                // (15,9): warning CS8602: Dereference of a possibly null reference.
+                //         field4.ToString(); // 2
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "field4").WithLocation(15, 9));
+        }
+
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/56256")]
+        public void MemberNotNull_InstanceLocalFunction_MissingArgument()
```

Comment:
>InstanceLocalFunction [](http://example.com/codeflow?start=34&length=21)

Consider calling these "NonStaticLocalFunction" since I don't think we have a term "instance local function".